26.md

```diff
@@ -0,0 +1,29 @@
+NIP-26
+======
+
+Relays List
+-----------
+
+`draft` `optional` `author:fiatjaf`
+
+A special event with kind `10001`, meaning "relay list" is defined as having a list of tags, one for each relay the author uses.
+
+The content is not used.
+
+The tags consist of arrays of 3 elements: the first is the relay URL, the second is the "read" condition -- which if set `"true"` means the client is reading events from that relay --, the third is the "write" condition -- which if set to `"true"` means the user is writing events to that relay.
```

Comment:
Instead of using "true" or "false", I think it's better to use a sequence of two strings: either `read` or `write` followed by `scope` which is runes-like.

- If you only want to read notes:
  `["wss://rss.relay.com", "read", "kind=1"]`

- If you only want to communicate with `aabb` over this relay:
  `["wss://private.relay.com", "read", "kind=4 & pubkey^aabb", "write", "kind=4 & p^aabb"]`

- If you only want to set metadata on this relay:
  `["wss://rss.relay.com", "write", "kind=0"]`

Read and write can repeat many times to add more scopes. The allow everything scope is an empty string.

Reply:
I like that.

Reply:
OK, I tried to implement your proposal and I think it is considerably more annoying to implement than just `["wss://relay.com", "kinds^4", "kind=4"]` (the first condition is the "read" one, the second is the "write" one). What is the downside of this?

Reply:
No downside. I think it's good.

Reply:
The downside is that there is no way to specify a condition that evaluates to `false` (while on your proposal you can just omit that attribute).

Reply:
![2022-08-13-194848_579x366_scrot](https://user-images.githubusercontent.com/1653275/184515922-cb51ab7e-9ea7-4023-9608-b722576fc11f.png)

Actually according to this the `#` can be used for `true` and then we can use `""` for `false`.

Reply:
Hmm, but that is no good as it conflicts with tag queries that begin with `#` very badly.
Also it means "a comment" which implies an empty string should be the same as a string with a comment on it, so it makes no sense to evaluate `""` differently than `"#"`.

We should probably change so the naked operator `"!"` means `false` and then we can use `""` to mean `true`.

Reply:
See my next proposal: https://github.com/nostr-protocol/nips/pull/32/commits/418cd859fcb12dd6d4ffc5a1e00880b6039b6c09

Reply:
`kind<0` seems hacky but would result in false